Lib/socket.py

```diff
@@ -591,7 +591,7 @@ def readinto(self, b):
                 self._timeout_occurred = True
                 raise
             except error as e:
-                if e.args[0] in _blocking_errnos:
+                if e.args and e.args[0] in _blocking_errnos:
```

Comment:
Why not use `e.errno`?

In what cases `e.args` is empty? Maybe this is an error which should be fixed?

Reply:
I check out the error Msg, but nothing is get.

But if i check the error type like bellow, 
<img width="425" alt="2018-02-02 3 23 37" src="https://user-images.githubusercontent.com/9625741/35719526-260f1320-082d-11e8-8bf4-87032fdc7da3.png">

It is ```TimeoutError``` which is not caught in previous timeout except clause.
<img width="389" alt="2018-02-02 3 26 19" src="https://user-images.githubusercontent.com/9625741/35719589-776bfd8c-082d-11e8-9dd5-166270e8aeff.png">

The empty case what you asked me is a TimeoutError not caught in timeout previous except clause

Reply:
Please don't use images for text information. They are hard to read by blind people and are disappeared in email clients.

Reply:
Thus the actual bug is that TimeoutError is not caught in previous except clause?

Reply:
Oh sorry for images.
and yes. you right.  TimeoutError not caught